src/vm/stublink.cpp

```diff
@@ -2201,6 +2205,14 @@ Stub* Stub::NewStub(PTR_VOID pCode, DWORD flags)
         pBlock = new (executable) BYTE[totalSize];
 #else
         pBlock = new BYTE[totalSize];
+#if defined(PLATFORM_UNIX) && defined(_ARM_)
+        _ASSERTE(totalSize <= 0x1000);
```

Comment:
This hack is really problematic from security point of view. It will flip random parts of the regular head to executable.